packages/core/onLongPress/index.ts

```diff
@@ -42,13 +50,41 @@ export function onLongPress(
 
   let timeout: ReturnType<typeof setTimeout> | undefined
   let posStart: Position | undefined
+  let startTimestamp: number | undefined
+  let hasLongPressed = false
 
   function clear() {
     if (timeout) {
       clearTimeout(timeout)
       timeout = undefined
     }
     posStart = undefined
+    startTimestamp = undefined
+    hasLongPressed = false
+  }
+
+  function onRelease(ev: PointerEvent) {
+    if (options?.modifiers?.self && ev.target !== elementRef.value)
+      return
+
+    if (!options?.onMouseUp)
+      return
```

Comment:
todo: early returns must also call `clear()`, or we could call clear early, and save a copy of variables that should be used in `options.onMouseUp()`